storage/src/main/java/org/apache/uniffle/storage/common/AbstractStorage.java

```diff
@@ -40,17 +41,22 @@ public abstract class AbstractStorage implements Storage {
   abstract ShuffleWriteHandler newWriteHandler(CreateShuffleWriteHandlerRequest request);
 
   @Override
-  public ShuffleWriteHandler getOrCreateWriteHandler(CreateShuffleWriteHandlerRequest request) {
+  public Pair<ShuffleWriteHandler, Boolean> getOrCreateWriteHandler(
```

Comment:
I don't hope to change the original abstract definition, this pair returning structure looks tricky.

From my sight, why not making the is_created or is_cached as the internal var into the concreted ShuffleWriteHandler, and then expose this as a interface method?

Reply:
I will recheck the code base on your suggestion.

Reply:
I rechecked the code, but i didn't find a better way to fix it. Can you give me some code level suggestions?
@zuston

Reply:
```
class ShuffleWriteHandlerWrapper {
  private ShuffleWriteHandler handler;
  private boolean isNewlyCreated;
}


@Override
  public ShuffleWriteHandler getOrCreateWriteHandler(CreateShuffleWriteHandlerRequest request) {
    writerHandlers.computeIfAbsent(request.getAppId(), key -> JavaUtils.newConcurrentMap());
    requests.computeIfAbsent(request.getAppId(), key -> JavaUtils.newConcurrentMap());
    Map<String, ShuffleWriteHandler> map = writerHandlers.get(request.getAppId());
    String partitionKey =
        RssUtils.generatePartitionKey(
            request.getAppId(), request.getShuffleId(), request.getStartPartition());
    map.computeIfAbsent(partitionKey, key -> newWriteHandler(request));
    Map<String, CreateShuffleWriteHandlerRequest> requestMap = requests.get(request.getAppId());
    requestMap.putIfAbsent(partitionKey, request);
    ShuffleWriteHandler handler = map.get(partitionKey);
    return new ShuffleWriteHandlerWrapper(handler, true/false);
  }
```

like this

Reply:
Fixed